pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/PulsarZooKeeperClient.java

```diff
@@ -296,6 +297,8 @@ protected PulsarZooKeeperClient(String connectString,
         this.connectExecutor =
                 Executors.newSingleThreadExecutor(
                         new ThreadFactoryBuilder().setNameFormat("ZKC-connect-executor-%d").build());
+        this.callbackExecutor = Executors.newCachedThreadPool(
```

Comment:
I think the logic of the executor belongs to the AbstractMetadataStore since it's applicable to all the implementations under it.